### DISCUSSION TITLE
bug: TypeError when reading contract

### DISCUSSION THREAD

**Discussion by aarifkhamdi** [wevm/viem#1928] [Question] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

2.7.22

### Current Behavior

Sometimes I'm getting the error

TypeError: Cannot read properties of null (reading 'data')
    at getRevertErrorData (/app/node_modules/.pnpm/viem@2.7.22_typescript@5.3.3_zod@3.22.4/node_modules/viem/actions/public/call.ts:359:54)
    at call (/app/node_modules/.pnpm/viem@2.7.22_typescript@5.3.3_zod@3.22.4/node_modules/viem/actions/public/call.ts:224:18)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at readContract (/app/node_modules/.pnpm/viem@2.7.22_typescript@5.3.3_zod@3.22.4/node_modules/viem/actions/public/readContract.ts:112:22)
    at ... 

### Expected Behavior

No "TypeError" here https://github.com/wevm/viem/blob/a388b7ceadfd1b94f6fb537e5ce47ec31cbebff7/src/actions/public/call.ts#L359

### Steps To Reproduce

1. clone https://github.com/aarifkhamdi/viem-reading-null
2. `pnpm I`
3. run this https://github.com/aarifkhamdi/viem-reading-null/blob/1a0cb805c7689f29f3a79a1eb1ad9e7aa27247d8/src/main.ts#L1

**Comment by aarifkhamdi** [comment]
@jxom why did you close the issue? How can I fix it by myself? I think it’s library bug. It does not handle errors properly

**Reply by jxom** [reply]
Hard to tell what's going on without a minimal reproducible example. Most likely could be a RPC provider that responds with a malformed response/response that isn't shaped properly.

**Reply by aarifkhamdi** [reply]
@jxom I've invited you to the example repo https://github.com/aarifkhamdi/viem-reading-null

**Reply by aarifkhamdi** [reply]
I added minimal reproducible example. Reopen issue please

**Reply by jxom** [reply, marked as answer]
https://github.com/wevm/viem/issues/1939
